lib/modules/manager/pip-compile/common.ts

```diff
@@ -27,12 +59,155 @@ export function getPipToolsConstraint(config: UpdateArtifactsConfig): string {
 
   return '';
 }
+
 export const constraintLineRegex = regEx(
-  /^(#.*?\r?\n)+# {4}pip-compile(?<arguments>.*?)\r?\n/,
+  /^(#.*?\r?\n)+# {4}(?<command>\S*)(?<arguments> .*?)?\r?\n/,
 );
-export const allowedPipArguments = [
+
+export const disallowedPipOptions = [
+  '--no-header', // header is required by this manager
+];
+export const optionsWithArguments = [
+  '--output-file',
+  '--extra',
+  '--extra-index-url',
+  '--resolver',
+];
+export const allowedPipOptions = [
+  '-v',
   '--allow-unsafe',
   '--generate-hashes',
-  '--no-emit-index-url',
+  '--no-emit-index-url', // TODO: handle this!!!
   '--strip-extras',
+  ...optionsWithArguments,
 ];
+
+// as commander.js is already used, we will reuse it's argument parsing capability
+const dummyPipCompile = new Command();
+dummyPipCompile
+  .argument('[sourceFile...]') // optional, so extractHeaderCommand can throw an explicit error
+  .option('--output-file <path>')
+  .option('--extra <extra...>')
+  .option('--extra-index-url <url...>')
+  .allowUnknownOption();
+
+// TODO(not7cd): test on all correct headers, even with CUSTOM_COMPILE_COMMAND
+export function extractHeaderCommand(
+  content: string,
+  fileName: string,
+): PipCompileArgs {
+  const strict: boolean = true; // TODO(not7cd): add to function params
+  const compileCommand = constraintLineRegex.exec(content);
+  if (compileCommand?.groups === undefined) {
+    throw new Error(`Failed to extract command from header in ${fileName}`);
+  }
+  logger.debug(
+    `pip-compile: found header in ${fileName}: \n${compileCommand[0]}`,
+  );
+  const command = compileCommand.groups.command;
+  const argv: string[] = [command];
```

Comment:
```suggestion
  const argv = [command];
```
I think we can skip the type here